core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -2741,6 +2755,18 @@ class ReplicaManager(val config: KafkaConfig,
     }
   }
 
+  private def mayUpdateTopicAssignment(partition: Partition, partitionDirectoryId: Option[Uuid]) = {
+    if (partitionDirectoryId.isDefined) {
+      val topicPartitionActualDirectory = partition.log.flatMap(log => logManager.directoryId(log.dir.getParent))
+      if (!topicPartitionActualDirectory.exists(partitionDirectoryId.contains)) {
+        topicPartitionActualDirectory
+          .flatMap(uuid => partition.topicId.map(topicId =>
+            directoryEventHandler.handleAssignment(new common.TopicIdPartition(topicId, partition.partitionId), uuid)
+          ))
```

Comment:
It is odd to map and flatMap but then throw away the result. Should this be a foreach instead?